### PR TITLE
Add sequence diagram and update note command in DG

### LINKED ISSUE #112: Docs: Add UML diagram for note command
Referencing #110.

### PR DESCRIPTION
**Pull request by ruijietay**
Resolves #112.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -204,6 +199,16 @@ The following activity diagram summarizes what happens when a QR Code is generat
 
 _vCard_ is a data format for contact information. Detailed information can be found in [RFC 6350](https://datatracker.ietf.org/doc/html/rfc6350).
 
+### Adding a Note To An Existing Client
+
+The sequence diagram below shows how the components interact with each other when the user inputs the command `note 1 note/Likes to eat`.
+
+![AddNoteSequenceDiagram](images/AddNoteSequenceDiagram.png)
```
Comment (RyanNgWH) [inline docs/DeveloperGuide.md:206]: Seems like the sequence diagram is missing

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -204,6 +199,16 @@ The following activity diagram summarizes what happens when a QR Code is generat
...
+
+The diagram highlights the four main components of FitBook, highlighted in their respective colors. For more information regarding the four main components, see [Main components of the architecture](#main-components-of-the-architecture).
```
Comment (RyanNgWH) [inline docs/DeveloperGuide.md:208]: The hyperlink here does not seem to be working. 

I believe that markdown can only natively link to headers and not bolded text. Maybe use a HTML anchor tag instead?

**Review by RyanNgWH — CHANGES_REQUESTED** [review]
There are some broken links that should be fixed. Otherwise, the descriptions look alright :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ruijietay) [reply docs/DeveloperGuide.md:206]: Good catch!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ruijietay) [reply docs/DeveloperGuide.md:208]: Good catch! Will fix.

**Review by RyanNgWH — APPROVED** [review]
Nice! All seems good and working now. Lets merge!

**Review by dillontkh — APPROVED** [review]
LGTM!

**Pull request merged**